### DISCUSSION TITLE
how to use `LayoutService`

### DISCUSSION THREAD

**Discussion by sdg32** [vmware/clarity#6603] [Q&A]
I want to create a new form input component,  but where can I import the `LayoutService` from and inject it to my component?

I see the `LayoutService` export as `ɵb` in `clr-angular.d.ts`,  can I use it?

**Comment by Jinnie** [comment]
The service itself is not exported. As a workaround you could in theory grab an instance from the ClrForm (it's public on the TS level). 
You will need to do it in/after AfterViewInit, though, which will lead to complicated initialization and change detection (the ExpressionChecked.... error in the demo below)
https://stackblitz.com/edit/layout-service-demo?file=src/app/app.component.ts
Not really clean, but some path forward.
